tracer/src/Datadog.Trace/Iast/Aspects/System/StringAspects.cs

```diff
@@ -512,6 +512,104 @@ public static string Concat<T>(IEnumerable<T> values)
         return result;
     }
 
+#if NET6_0_OR_GREATER
+
+    /// <summary>
+    /// String.Join aspect
+    /// </summary>
+    /// <param name="param1"> First param </param>
+    /// <param name="param2"> Second param </param>
+    /// <returns> Join result </returns>
+    [AspectMethodReplace("System.String::Concat(System.ReadOnlySpan`1<System.Char>,System.ReadOnlySpan`1<System.Char>)")]
+    public static string Concat(ReadOnlySpan<char> param1, ReadOnlySpan<char> param2)
+    {
+        var result = string.Concat(param1, param2);
+        try
+        {
+            return StringModuleImpl.OnStringConcat(TaintedObjects.GetString(ref param1), TaintedObjects.GetString(ref param2), result);
+        }
+        catch (Exception ex)
+        {
+            IastModule.LogAspectException(ex, $"{nameof(StringAspects)}.{nameof(Join)}");
```

Comment:
I guess the method reported in the error here should be `Concat` instead of `Join`?

Reply:
```suggestion
            IastModule.LogAspectException(ex, $"{nameof(StringAspects)}.{nameof(Concat)}");
```

Reply:
Fixing it